mario-karen/server.js

```diff
@@ -0,0 +1,184 @@
+  // This is 2 and 5. app.get above is a event listener-like method listening for /new-article and once there is a request, it will tell send back new.html in the line of code below. It is not interacting with any article.js. SO the user probably has to type /new-article in the URL to access this. None of the CRUD is enacted.
+  response.sendFile('new.html', {root: './public'});
+});
+
+
+// REVIEW: Routes for making API calls to use CRUD Operations on our database
+app.get('/articles', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This is 2,3,4, and 5. app.get is the 2, client.query is the 3 and its results that it sends back is the 4 and the response.send is the 5. It is interacting with fetchAll() from article.js which is the request that that the server is listening for. It is the READ from CRUD.
+  client.query('SELECT * FROM articles;')
+    .then(function(result) {
+      response.send(result.rows);
+    })
+    .catch(function(err) {
+      console.error(err)
+    })
+});
+
+app.post('/articles', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This is 2,3,4,5 again as well. 2 is the request the app.post is listening for. Client.query is the 3 where the server is telling the DB to do stuff(in this case insert a new row) and the data base tells the server back that its done it would be 4 and response.send('insert complete') would be 5. This is interacting with insertRecord in article.js and it is the Update in CRUD
+  let SQL = `
+    INSERT INTO articles(title, author, "authorUrl", category, "publishedOn", body)
+    VALUES ($1, $2, $3, $4, $5, $6);
+  `;
+
+  let values = [
+    request.body.title,
+    request.body.author,
+    request.body.authorUrl,
+    request.body.category,
+    request.body.publishedOn,
+    request.body.body
+  ]
+
+  client.query( SQL, values )
+    .then(function() {
+      response.send('insert complete')
+    })
+    .catch(function(err) {
+      console.error(err);
+    });
+});
+
+app.put('/articles/:id', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // 2345 with CRUD as Update. interacting with updateRecord.
+
+  let SQL = 'UPDATE articles SET title = $2, author =$3, "authorUrl" = $4, category = $5, "publishedOn" = $6, body = $7 WHERE article_id=$1;';
```

Comment:
For your own sanity, it's nice to be consistent in spacing around equals signs (i.e. for setting author)